### DISCUSSION TITLE
Navigation fails when scripts are blocked or not loaded

### DISCUSSION THREAD

**Discussion by IsraelOrtuno** [nuxt/nuxt#16249] [Questions] [answered]
I am having some problems with some AdBlockers which made me consider how nuxt navigation works. The AdBlocker is preventing loading that particular file for whatever reason (probably because its name "SocialLinks") when I navigate to a certain page. Since that component is required or used in the next page, the navigation fails as the client is not able to load that file. I wonder if anyone knows how to deal with this situation where the load fails and the client basically "does nothing" as it won't load the next route and I guess this may be a common scenario since ad blockers are widely used.

<img width="956" alt="image" src="https://user-images.githubusercontent.com/1769417/204639823-5a690e74-4b34-4088-8099-77c8a16b9189.png">

**Comment by TheAlexLichter** [comment, marked as answer]
Even though it has another root cause, https://github.com/nuxt/framework/issues/6648 sounds like an option here.

**Reply by IsraelOrtuno** [reply]
Yeah the AdBlocker is just a use case, same thing with deployments changing filenames I guess, will keep an eye to that issue. Thanks a lot 👍

Also got this tip from Discord for anyone facing issues with AdBlockers in the future:
https://discord.com/channels/473401852243869706/897487139888062506/1047278494180184084

> Watt — Yesterday at 22:02
> here: https://easylist.to/easylist/fanboy-social.txt probably matches:
> @@||nuxt.com/_nuxt/SocialLinks.$script,~third-party
> I would just rename the component/module. I had a similar problem when naming dom elements like "banner"

This seems to be a list of kind of regexes for identifying potential ads.
